prefect_bitbucket/credentials.py

```diff
@@ -75,6 +77,23 @@ def _validate_username(cls, value: str) -> str:
             raise ValueError("Username cannot be longer than 30 chars.")
         return value
 
+    @validator("token", pre=True, always=True)
+    def _validate_token(cls, value: Optional[SecretStr]) -> Optional[SecretStr]:
```

Comment:
should this be `Optional[Union[SecretStr, str]]` because you're doing `value.get_secret_value() if isinstance(value, SecretStr) else value`?

Reply:
Hm, actually since we should only expect a `SecretStr`, I've simplified the logic. I've also updated a test that had failed due to passing in a `str` token.